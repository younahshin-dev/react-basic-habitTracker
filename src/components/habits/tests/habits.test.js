import '@testing-library/jest-dom'
import Habits from '../habits';
import renderer from 'react-test-renderer';

describe('Habits', () => {

  const habits = [
    {id: 1, name: 'Reading', count:0},
    {id: 2, name: 'Running', count:1},
    {id: 3, name: 'Coding', count:0}
  ]; 

  it('renders', () => {
    //스냅샷 테스트
    const component = renderer.create(
      <Habits
         habits={habits}
         onIncrement={jest.fn()}
         onDecrement={jest.fn()}
        onDelete={jest.fn()}
        onAdd={jest.fn()}
        onReset={jest.fn()}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
