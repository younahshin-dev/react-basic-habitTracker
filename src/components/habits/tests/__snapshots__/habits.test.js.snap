// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habits renders 1`] = `
<div
  className="habits"
>
  <form
    className="add-form"
    onSubmit={[Function]}
  >
    <input
      className="add-input"
      placeholder="Habit"
      type="text"
    />
    <button
      className="add-button"
    >
      Add
    </button>
  </form>
  <ul>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        Reading
      </span>
      <span
        className="habit-count"
      >
        0
      </span>
      <button
        className="habit-button habit-increase"
        data-testid="increseBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        data-testid="decreseBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        data-testid="deleteBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        Running
      </span>
      <span
        className="habit-count"
      >
        1
      </span>
      <button
        className="habit-button habit-increase"
        data-testid="increseBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        data-testid="decreseBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        data-testid="deleteBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        Coding
      </span>
      <span
        className="habit-count"
      >
        0
      </span>
      <button
        className="habit-button habit-increase"
        data-testid="increseBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        data-testid="decreseBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        data-testid="deleteBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
  </ul>
  <button
    className="habits-reset"
    onClick={[MockFunction]}
  >
    Reset All
  </button>
</div>
`;
